internals/utils/status/capp.go

```diff
@@ -119,6 +122,31 @@ func SyncStatus(ctx context.Context, capp rcsv1alpha1.Capp, log logr.Logger, r c
 	}
 	cappObject.Status.KnativeObjectStatus = kservice.Status
 	cappObject.Status.RevisionInfo = RevisionsStatus
+	if cappObject.Spec.LogSpec != (rcsv1alpha1.LogSpec{}) {
```

Comment:
Take all the flow you did here and separate it to different function`buildLoggingStatus` this function will return a status object. make sure you don't pass to wide parameters there's no need to pass all the capp object pass only the relevant parameters. In general its important to make our functions short and preserve the single responsibility principle.